docs/python-minifier-analysis.md

```
@@ -0,0 +1,134 @@
+
+Annotations are preserved for classes that require them:
+
+- `@dataclass` decorated classes
+- Classes inheriting from `typing.NamedTuple`
+- Classes inheriting from `typing.TypedDict`
+
+Detection uses decorator and base class inspection via the parent-tracked AST.
+
+### 4. Implementation Details
+
+#### Key Methods:
```

markdownlint-cli2 (0.17.2)

Trailing punctuation in heading
Punctuation: ':'

(MD026, no-trailing-punctuation)